source/base/data_out_base.cc

```diff
@@ -8685,13 +8690,26 @@ namespace
                                  node_dataspace,
                                  H5P_DEFAULT);
 #  else
-        node_dataset    = H5Dcreate(h5_mesh_file_id,
+        node_dataset_id = H5Pcreate(H5P_DATASET_CREATE);
+#    ifdef DEAL_II_WITH_ZLIB
+        H5Pset_deflate(node_dataset_id,
+                       get_zlib_compression_level(flags.compression_level));
+        H5Pset_chunk(node_dataset_id, 2, node_ds_dim);
+#    else
+        AssertThrow(
```

Comment:
You set the default to best speed but throw an error here if no zlib support is found, right? That is not a good default behavior!

Reply:
I would like to avoid a situation where we break user code. I think for vtu we ignore the compression flag if zlib is not found (instead of an error). That seems like a better solution to me. That way you can set the default to enable compression.

Reply:
Yes, I think ignoring the flag when no zlib is available is the right way to deal with the situation.